doc/source/dev/contributor/rendering_documentation.rst

```diff
@@ -217,6 +217,70 @@ Some examples:
         Some text that follows the list.
 
 
+Self-contained examples
+~~~~~~~~~~~~~~~~~~~~~~~
+All examples must be self-contained. It means all imports must be explicit,
+the data used must be defined in the examples, etc.
+
+    Yes::
+
+        >>> import numpy as np
+        >>> rng = np.random.default_rng()
+
+    No::
+
+        >>> rng = np.random.default_rng()
+
+The scope is defined
+per docstrings. i.e. for a given function's docstrings, it's possible,
+and a good thing, to separate code with explanations. Note that blank lines
+surrounding all block codes are required.
+
+    Yes::
+
+        Some initial text
+
+        >>> import numpy as np
+        >>> rng = np.random.default_rng()
+
+        This is some explanation
+
+        >>> rng.random(10)
+
+
+Examples and randomness
+~~~~~~~~~~~~~~~~~~~~~~~
+Examples are verified in the CI and output is compared. The main goal
+is to ensure that the API and usage are consistent. Doctests are not unittests.
+
+In case a random number generator is needed, `np.random.Generator` must be
```

Comment:
Would it be wrong to state that `np.random.default_rng` must be used to generate one? This would be more obvious from the example below.